urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/httpd/ContentRange.java

```diff
@@ -0,0 +1,261 @@
+     **/
+    public ContentRange( String bytesUnit,
+			 long firstBytePos,
+			 long lastBytePos,
+			 long instanceLen ) 
+	throws NullPointerException,
+	       IllegalArgumentException {
+
+	super();
+
+	if( bytesUnit == null )
+	    throw new NullPointerException( "Cannot create Content-Range wrappers with null-bytesUnit." );
+	if( !bytesUnit.equalsIgnoreCase(ContentRange.NAME_BYTESUNIT_BYTES) )
+	    throw new IllegalArgumentException( "Unknown/unsupported value for bytesUnit: " + bytesUnit + "." );
+	if( firstBytePos < 0 )
+	    throw new IllegalArgumentException( "firstBytePosition must not be less than zero (" + firstBytePos + ")." );
+	if( instanceLen >= 0 ) {
+
+	    if( firstBytePos >= instanceLen )
+		throw new IllegalArgumentException( "firstBytePosition cannot be larger than instanceLength (" + firstBytePos + " >= " + instanceLen + ")." );
+
+	    if( lastBytePos >= instanceLen )
+		throw new IllegalArgumentException( "lastBytePosition cannot be larger than instanceLength (" + lastBytePos + " >= " + instanceLen + ")." );
+
+	}
+	if( firstBytePos > lastBytePos )
+	    throw new IllegalArgumentException( "firstBytePosition must not be larger than lastBytePosition (" + firstBytePos + " > " + lastBytePos + ")." );
+
+
+	this.bytesUnit         = bytesUnit;
+	this.firstBytePosition = firstBytePos;
+	this.lastBytePosition  = lastBytePos;
+	this.instanceLength    = instanceLen;
+
+    }
+
+    /**
+     * Get the 'bytes-unit' value.
+     * Currently the only supported value is "bytes".
+     *
+     * @return The value of the 'bytes-unit'; the returned string is never null.
+     **/
+    public String getBytesUnit() {
+	return this.bytesUnit;
+    }
+
+    /**
+     * Get the 'first-byte-position' value.
+     *
+     * @return The value of 'first-byte-position'; the returned value is never negative.
+     **/
+    public long getFirstBytePosition() {
+	return this.firstBytePosition;
+    }
+
+    /**
+     * Get the 'last-byte-position' value.
+     *
+     * @return The value of 'last-byte-position'; the returned value is never negative
+     *         and always greater than or equal to 'first-byte-position'.
+     **/
+    public long getLastBytePosition() {
+	return this.lastBytePosition;
+    }
+     
+    public long calculateLength() {
+	return this.lastBytePosition - this.firstBytePosition + 1;
+    }
+
+    /**
+     * Get the 'instance-length' value; the returned value is larger than or equal to
+     * 'last-byte-position' OR it is -1L, which means that the instance-length is currently
+     * unknown.
+     *
+     * @return The value of 'instance-length'.
+     **/
+    public long getInstanceLength() {
+	return this.instanceLength;
+    }
+
+
+    /**
+     * This static method parses the given 'Content-Range' header field value.
+     * The passed string must not be null and match the specification in RFC 2616/sec14.
+     **/ 
+    public static ContentRange parse( String headerValue ) 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 211-234 into a separate method.